### PR TITLE
Monster Shrine & Material Storage update

### LINKED ISSUE #178: Monsters in Monster Sealed Shrine not visible in Swarfarm
After yesterday's update (v6.1.0) and introduction of new bulding (Monster Sealed Shrine), if monsters are stored in this building, these are no longer visible in Swarfarm.

I did some digging in JSON Profile file we get from SWEX and - sadly - these monsters are no longer in `unit_list` key. I thought it works similar to Material Storage Building, when we can store units like Rainbowmons. New building has ID `83` (*building_master_id*), but sadly there's no other reference. 

Also, there's new command `getUnitStorageList`. I think its response contains what we are looking for, since it has `unit_master_id` same as monster `unit_master_id` from existing Monster objects, together with `quantity` and `class`.
[Example request](https://pastebin.com/CWq8wGYB)
[Example response](https://pastebin.com/G2C7nRFJ)

I don't think there's anything we can do, unless merging `HubUserLogin` command with `getUnitStorageList` at the built-in Profile plugin level, so people using webpage can import whole monster collection, not only the ones in inventory and normal storage.

### PR DESCRIPTION
**Pull request by QuatZo**
Reworked Material Storage & added Monster Shrine Storage
resolves #178

### CODE REVIEW

**Comment by QuatZo** [comment]
Everything should be done.

Some things to do after (or before) merging it to master:
* Parse new game data and replace Monster Shrine image (I just used placeholder from my phone's screenshot); sadly I can't get new game data with images right now
* Check Material Storage images and - if needed - replace names (I've seen different values in DB & Fixtures)

File: herders/tasks.py [hunk 1]
```diff
@@ -48,65 +50,55 @@ def com2us_data_import(data, user_id, import_options):
...
-        summoner.storage.rainbowmon_2_20 = results['inventory'].get('rainbowmon_2_20', 0)
-        summoner.storage.rainbowmon_3_1 = results['inventory'].get('rainbowmon_3_1', 0)
-        summoner.storage.rainbowmon_3_25 = results['inventory'].get('rainbowmon_3_25', 0)
-        summoner.storage.rainbowmon_4_1 = results['inventory'].get('rainbowmon_4_1', 0)
-        summoner.storage.rainbowmon_4_30 = results['inventory'].get('rainbowmon_4_30', 0)
-        summoner.storage.rainbowmon_5_1 = results['inventory'].get('rainbowmon_5_1', 0)
-
-        summoner.storage.save()
+        # inventory bulk update or create
+        all_inv_items = {gi.com2us_id: gi for gi in GameItem.objects.all()}
```
Comment (QuatZo) [inline herders/tasks.py]: There should be category__isnull=False filter

Comment (QuatZo) [reply herders/tasks.py]: Done

**Comment by QuatZo** [comment]
I forgot about Fusion system. WIP.

**Comment by QuatZo** [comment]
Fusion system adjusted to newest Models.

**Comment by PeteAndersen** [comment]
I pushed a few small changes based on my testing/preferences. The only remaining issue is that this wipes out a user's existing storage when the migration executes. Many users do not regularly import or use SWEX so this would be a hassle for them. The migration should update the new models with the data from the old model. 

I believe the best way to execute this is to split the migration into 3 unique stages:

1. Create new storage models
2. Data migration
3. Delete old storage migration

The data migration should be able to pick up where it left off in case it fails to complete in one pass. That probably means some way of filtering a queryset down to only storages that still need to be migrated.

**Review by PeteAndersen — APPROVED** [review]

**Pull request merged**